presto-parser/src/test/java/com/facebook/presto/sql/parser/TestSqlParser.java

```diff
@@ -1628,6 +1628,44 @@ public void testShowColumnStats()
             assertStatement(format("SHOW STATS %s", qualifiedName), new ShowStats(qualifiedName));
             assertStatement(format("SHOW STATS FROM %s", qualifiedName), new ShowStats(qualifiedName));
             assertStatement(format("SHOW STATS IN %s", qualifiedName), new ShowStats(qualifiedName));
+
+            assertStatement(format("SHOW STATS FROM %s WHERE row_count=17", qualifiedName),
+                    new ShowStats(qualifiedName,
+                            Optional.of(new ComparisonExpression(ComparisonExpression.Type.EQUAL, new QualifiedNameReference(QualifiedName.of("row_count")), new LongLiteral("17"))),
+                            ImmutableList.of(),
+                            Optional.empty()));
+            assertStatement(format("SHOW STATS FROM %s WHERE row_count=17 ORDER BY column_name", qualifiedName),
+                    new ShowStats(qualifiedName,
+                            Optional.of(new ComparisonExpression(ComparisonExpression.Type.EQUAL, new QualifiedNameReference(QualifiedName.of("row_count")), new LongLiteral("17"))),
+                            ImmutableList.of(new SortItem(new QualifiedNameReference(QualifiedName.of("column_name")), SortItem.Ordering.ASCENDING, SortItem.NullOrdering.UNDEFINED)),
+                            Optional.empty()));
+            assertStatement(format("SHOW STATS FROM %s WHERE row_count=17 ORDER BY column_name LIMIT 18 ", qualifiedName),
+                    new ShowStats(qualifiedName,
+                            Optional.of(new ComparisonExpression(ComparisonExpression.Type.EQUAL, new QualifiedNameReference(QualifiedName.of("row_count")), new LongLiteral("17"))),
+                            ImmutableList.of(new SortItem(new QualifiedNameReference(QualifiedName.of("column_name")), SortItem.Ordering.ASCENDING, SortItem.NullOrdering.UNDEFINED)),
+                            Optional.of("18")));
+
```

Comment:
Because I like it that way and it doesn't break coding standard :>
Also, each group have different pattern of additional parameters.